### PR TITLE
Enforce a few more rubocop guidelines, minor stylistic fixups

### PR DESCRIPTION
**Pull request by Amorymeltzer**
1. Only two casks (`java` and `applepi-baker`) had `:cookies` on three lines, moved 'em to one line for the sake of convenience and matching. (Technically not rubocop, but felt silly doing a PR for this by it's lonesome)<sup>1</sup>
2. Rubocop: Removed whitespace following the cask header in 29 casks, should enforce `no_empty_lines` in `Style/EmptyLinesAroundBlockBody`;   @jawshooah You were originally unsure if this was worthwhile, I rather think it's a good idea, if only for regex/parsing.
3. Rubocop: Tidied up casks based on Style/AlignHash and Style/AlignParameters (also one each from Style/CommentIndentation<sup>2,3</sup> and Style/EmptyLineBetweenDefs)
4. Rubocop: Removed spaces found inside square brackets, should enforce `Style/SpaceInsideBrackets`
5. ~~Rubocop: Enforce `Style/Tab` and `Style/TrailingWhitespace` (no current errors, intended from the start in #9921, they are indeed "gross")~~

Ping @jawshooah via #9921.

---

1: I couldn't get `applepi-baker` to fail without it, though, so perhaps it's no longer necessary?
2: Comments in `soulver` and `gpgtools` are waiting on glob support (#12991; see also #8202).
3: Comments in `ableton-*` are user-generated content being kept until `:trash` is properly implemented per https://github.com/caskroom/homebrew-cask/pull/11587#discussion_r31761341

### CODE REVIEW

**Comment by jawshooah** [comment]
I was actually waiting for them to release `0.36.0` before doing this, since there will be a couple of upcoming [additions](https://github.com/bbatsov/rubocop/pull/2529) and [fixes](https://github.com/bbatsov/rubocop/issues/2565) that will make things much nicer for us.

File: .rubocop.yml [hunk 1]
```diff
@@ -33,3 +36,12 @@ Style/FileName:
 
 Style/HashSyntax:
   EnforcedStyle: hash_rockets
+
+Style/SpaceInsideBrackets:
+  Enabled: true
+
+Style/Tab:
+  Enabled: true
```
Comment (jawshooah) [inline .rubocop.yml]: This is [enabled by default](https://github.com/bbatsov/rubocop/blob/v0.35.1/config/enabled.yml#L726-L729), no need to make it explicit.

File: .rubocop.yml [hunk 2]
```diff
@@ -33,3 +36,12 @@ Style/FileName:
 
 Style/HashSyntax:
   EnforcedStyle: hash_rockets
+
+Style/SpaceInsideBrackets:
+  Enabled: true
```
Comment (jawshooah) [inline .rubocop.yml]: [Same here.](https://github.com/bbatsov/rubocop/blob/v0.35.1/config/enabled.yml#L667-L670)

File: .rubocop.yml [hunk 3]
```diff
@@ -23,6 +23,9 @@ Performance/StringReplacement:
 Style/BarePercentLiterals:
   EnforcedStyle: percent_q
 
+Style/EmptyLinesAroundBlockBody:
+  EnforcedStyle: no_empty_lines
```
Comment (jawshooah) [inline .rubocop.yml]: [Same here.](https://github.com/bbatsov/rubocop/blob/v0.35.1/config/default.yml#L378-L379)

File: .rubocop.yml [hunk 4]
```diff
@@ -33,3 +36,12 @@ Style/FileName:
...
   EnforcedStyle: hash_rockets
+
+Style/SpaceInsideBrackets:
+  Enabled: true
+
+Style/Tab:
+  Enabled: true
+
+Style/TrailingWhitespace:
+  Enabled: true
```
Comment (jawshooah) [inline .rubocop.yml]: [Same here.](https://github.com/bbatsov/rubocop/blob/v0.35.1/config/enabled.yml#L741-L744)

**Comment by jawshooah** [comment]
All that aside, the changes look fine (and won't prevent us from doing more once we get a new RuboCop release), except for one thing: `AlignHash`. I'm partial to the `table` style (align keys with keys, hashrockets with hashrockets, and values with values). Not a big deal either way, but we should come to a consensus on that before merging.

@vitorgalvao, you're a graphic designer (sorry, I stalked your website), so you probably have a better eye for this than me. What would you suggest? (examples of the different styles are [here](https://github.com/bbatsov/rubocop/blob/v0.35.1/config/default.yml#L75-L88)).

**Comment by Amorymeltzer** [comment]
@jawshooah My thinking with the `AlignHash` changes was to just get those nine casks in line with the rest of them; the issue for them all was key alignment being way out of whack.

Personally I prefer the `key` design — less mucking about with spaces and aligning — but I was saving that conversation for later, to be had alongside `Style/SingleSpaceBeforeFirstArg` (per your comments at #9921)

**Comment by vitorgalvao** [comment]
> (sorry, I stalked your website)

Well, it is public after all, but it is in dire need of an update in every respect (the website itself and the works in it). It was the first html/css (as opposed to flash) website I’ve made, though, so with that mind I guess it isn’t *that* bad (well, a bit, I should really get to it eventually).

As for layout, I’d say `separator` is the only “wrong” one. When we have a list of things, having them start at different points (since we’re reading left-to-right and they’re aligned right-to-left) makes it harder to scan with good comprehension.

Personally (as in a matter of taste) I *like* `key` better, but I think `table` *works* better for reading/skimming/comprehension and that is more important. Users seem to be somewhat partial to it when submitting, as well. [Autocorrect support](https://github.com/caskroom/rubocop-cask/issues/1#issue-124013830) should take care of @Amorymeltzer’s concerns (i.e. no need to manually muck about with spacing) and popular text editors have plugins for this type of alignment. I say we go with `table`.

On a tangential note, I remember you (@jawshooah) asking about (months ago) why the heck do we only leave one space when indenting certain things:

```ruby
:symbol => [
            'one',
            'two'
           ]
```

That is indeed a weird pattern, and I’d be in favour of changing that to two spaces.

**Comment by Amorymeltzer** [comment]
I meant it more for contributors, manually spacing to make travis happy, but that is sort of the point of this whole thing.

One space is weird, weird, and so weird (speaking of manually spacing).

**Comment by jawshooah** [comment]
@vitorgalvao that particular oddity can't be fully rectified until a new release including [this PR](https://github.com/bbatsov/rubocop/pull/2529) is pushed.

In any case, I say we wait on this for now. @bbatsov has alluded to a release coming out in the next week or so.

**Comment by Amorymeltzer** [comment]
Mooted via #16317

**Pull request closed without merging**